library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java

```diff
@@ -47,7 +47,8 @@ public DownloadInfo newDownloadInfo(
                 SystemFacade systemFacade,
                 StorageManager storageManager,
                 DownloadNotifier notifier,
-                DownloadClientReadyChecker downloadClientReadyChecker) {
+                DownloadClientReadyChecker downloadClientReadyChecker,
+                Downloads downloads) {
```

Comment:
perhaps we should make the UriProvider now, might make the diff larger but it would be easier to review